gabbi/handlers/jsonhandler.py

```diff
@@ -95,7 +96,10 @@ def action(self, test, path, value=None):
                 rhs_path = test.replace_template('$' + rhs_path)
             info = test.load_data_file(value.replace('<@', '', 1))
             info = six.text_type(info, 'UTF-8')
-            value = self.loads(info)
+            if value.endswith('yaml'):
+                value = yaml.safe_load(info)
```

Comment:
You say in the commit message that you can't use solely `yaml.self_load()` because of the NanChecker, but it is not clear how that is related to this.

Reply:
Yeah good call, Looking at this again, I believe this has to do with a bug on how we test for isnan in tests/util.py we should cast the local var 'other' as a float because NaN is an IEEE 754 special value
(Pdb) other
'NaN'
(Pdb) math.isnan(other)
*** TypeError: a float is required